components/greaselion/browser/greaselion_service_impl.cc

```diff
@@ -230,6 +229,8 @@ GreaselionServiceImpl::GreaselionServiceImpl(
       update_pending_(false),
       pending_installs_(0),
       task_runner_(std::move(task_runner)),
+      extension_dirs_(new std::vector<base::ScopedTempDir>,
```

Comment:
you can't do this because `extension_dirs_` is accessed on the UI thread in `PostConvert`. In the destructor for GreaseLionServiceImpl you need 
```
task_runner_->DeleteSoon(FROM_HERE, std::move(extension_dirs_))
```
but I'm also concerned about this usage of ScopedTempDir in general because it's not immediately obvious that we're using this to clean up directories that are actually created somewhere else. I would prefer to just maintain a list of the directory paths and post a task in the destructor to delete them cc @pilgrim-brave

Reply:
I guess because of the way this works the `OnTaskRunnerDeleter` is actually maybe ok here, but I'd really like to change this to not used `ScopedTempDir` as mentioned above

Reply:
cc @emerick

Reply:
Good point, fixed in 1d261c302af03cc4965487defabeac5857de2839